mypyc/rt_subtype.py

```diff
@@ -51,5 +51,12 @@ def visit_rtuple(self, left: RTuple) -> bool:
                 is_runtime_subtype(t1, t2) for t1, t2 in zip(left.types, self.right.types))
         return False
 
+    def visit_rstruct(self, left: RStruct) -> bool:
+        if isinstance(self.right, RStruct):
+            return (self.right.name == left.name and self.right.names == left.names and
+                    len(self.right.types) == len(left.types) and all(is_runtime_subtype(t1, t2)
+                    for t1, t2 in zip(left.types, self.right.types)))
```

Comment:
I think that this should actually only compare equal if the other struct is the same as the first one, since there must only be one definition for each struct with a particular name, and C structs don't support structural compatibility.